thewalrus/_hafnian.py

```diff
@@ -261,17 +261,16 @@ def f_loop_odd(E, AX_S, XD_S, D_S, n, oddloop, oddVX_S):  # pragma: no cover
     Returns:
         array: polynomial coefficients
     """
-    E_k = E.copy()
 
     count = 0
     comb = np.zeros((2, n + 1), dtype=np.complex128)
```

Comment:
Since now the power traces inherit the type of your matrix, you don't need to make this complex, I believe. Worth checking this in a separate branch before doing the change here.